app/scripts/controllers/transactions/index.js

```diff
@@ -1084,6 +1150,83 @@ export default class TransactionController extends EventEmitter {
     }
   }
 
+  async confirmExternalTransaction(txMeta, txReceipt, baseFeePerGas) {
+    // add external transaction
+    await this.txStateManager.addExternalTransaction(txMeta);
+
+    const txId = txMeta.id;
+
+    if (!txMeta) {
+      return;
+    }
+
+    try {
+      // It seems that sometimes the numerical values being returned from
+      // this.query.getTransactionReceipt are BN instances and not strings.
+      const gasUsed =
+        typeof txReceipt.gasUsed === 'string'
```

Comment:
Strings have a `toString` method in JavaScript, so I don't think we need the type check.

Reply:
I copied these lines from the code for `confirmTransactions` above.

Reply:
But are you saying to just call `.toString(16)` on all types for `txReceipt.gasUsed`?

Reply:
The comments on lines 1179 and 1180 explain why the type check is needed

Reply:
But we could move some of the identical code between this method and the `confirmTransaction` method to a shared helper method. For example, we could create a `getNormalizedTxReceiptGasUsed` method